### PR TITLE
Fix upload command not working as expected when certain flags are used

### PR DESCRIPTION
**Pull request by silvanocerza**
**Please check if the PR fulfills these requirements**

- [x] The PR has no duplicates (please search among the [Pull Requests](https://github.com/arduino/arduino-cli/pulls)
      before creating one)
- [x] The PR follows
      [our contributing guidelines](https://arduino.github.io/arduino-cli/latest/CONTRIBUTING/#pull-requests)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] `UPGRADING.md` has been updated with a migration guide (for breaking changes)

* **What kind of change does this PR introduce?**

Fixes two bugs with a CLI command.

- **What is the current behavior?**

`upload` fails if the `--input-dir` or `--input-file` are used and Sketch path is not specified.
`upload` fails if `--fqbn` is not set even if specified in a `sketch.json` file.

* **What is the new behavior?**

`upload` doesn't fail `--input-dir` or `--input-file` are set and no Sketch path is specified.
`upload` doesn't fail if `--fqbn` is not specified and `sketch.json` contains a valid FQBN.

- **Does this PR introduce a breaking change, and is
[titled accordingly](https://arduino.github.io/arduino-cli/latest/CONTRIBUTING/#breaking)?**

Nope.

* **Other information**:

To verify that this PR fixes the aforementioned issue the integration tests must be run locally with a board connected. 
Arduino Uno and Arduino MKR WiFi 1010 are recommended.

---

See [how to contribute](https://arduino.github.io/arduino-cli/latest/CONTRIBUTING/)

### CODE REVIEW

File: cli/upload/upload.go [hunk 1]
```diff
@@ -81,27 +82,40 @@ func run(command *cobra.Command, args []string) {
 	if len(args) > 0 {
 		path = args[0]
 	}
-	sketchPath := arguments.InitSketchPath(path)
+	var sketchPath *paths.Path
+	var sk *sketch.Sketch = nil
+	// If the user explicitly specified a directory that contains binaries to upload
+	// use those, we don't really need a valid Sketch to upload in this case
```
Comment (per1234) [inline cli/upload/upload.go:88]: Do we care about this use case (which used to be supported), where the sketch argument could be used to provide `sketch.json` metadata when using a binary?:
```
$ ./arduino-cli version
arduino-cli.exe alpha Version: 0.18.3 Commit: d710b642 Date: 2021-05-14T12:36:58Z

$ ./arduino-cli sketch new /tmp/FooSketch
Sketch created in: C:\Users\per\AppData\Local\Temp\FooSketch

$ ./arduino-cli board attach arduino:avr:uno /tmp/FooSketch
Selected fqbn: arduino:avr:uno

$ ./arduino-cli compile --output-dir /tmp/FooSketchBin /tmp/FooSketch
Sketch uses 444 bytes (1%) of program storage space. Maximum is 32256 bytes.
Global variables use 9 bytes (0%) of dynamic memory, leaving 2039 bytes for local variables. Maximum is 2048 bytes.

$ ./arduino-cli upload -v -p COM10 --input-dir /tmp/FooSketchBin /tmp/FooSketch
"C:\program-files\arduino\cli\arduino-cli_nightly\directories-data\packages\arduino\tools\avrdude\6.3.0-arduino17/bin/avrdude" "-CC:\program-files\arduino\cli\arduino-cli_nightly\directories-data\packages\arduino\tools\avrdude\6.3.0-arduino17/etc/avrdude.conf" -v -V -patmega328p -carduino "-PCOM10" -b115200 -D "-Uflash:w:C:/Users/per/AppData/Local/Temp/FooSketchBin/FooSketch.ino.hex:i"

avrdude: Version 6.3-20190619
         Copyright (c) 2000-2005 Brian Dean, http://www.bdmicro.com/

[...]
```

Comment (silvanocerza) [reply cli/upload/upload.go:88]: Oh right, didn't think about that case. 🤔 
I'll try to fix it.

Comment (silvanocerza) [reply cli/upload/upload.go]: Fixed with 02539da.

**Review by per1234 — APPROVED** [review]
Fixes all three types of uploading issue for me.

**Pull request merged**